client/app/bundles/course/material/folders/handles.ts

```diff
@@ -1,20 +1,35 @@
+import { defineMessages } from 'react-intl';
+import { AxiosError } from 'axios';
+import { FolderData } from 'types/course/material/folders';
 import { getIdFromUnknown } from 'utilities';
 
 import CourseAPI from 'api/course';
 import { CrumbPath, DataHandle } from 'lib/hooks/router/dynamicNest';
 
+const translations = defineMessages({
+  folderNotFound: {
+    id: 'course.material.folders.FolderShow.folderNotFound',
+    defaultMessage: 'Folder not found',
+  },
+});
+
 const getFolderTitle = async (
   courseUrl: string,
   folderId?: number,
 ): Promise<CrumbPath> =>
   CourseAPI.folders
     .fetch(folderId)
     .then((response) => response.data.breadcrumbs)
+    .catch((error) => {
+      const response = (error as AxiosError<FolderData>).response;
+      if (!response?.data.breadcrumbs) throw new Error('Root folder not found');
+      return [...response.data.breadcrumbs, { id: -1, name: '' }];
+    })
     .then((breadcrumbs) => ({
       activePath: `${courseUrl}/materials/folders/${breadcrumbs[0].id}`,
       content: breadcrumbs.map((crumb) => ({
-        title: crumb.name,
-        url: `materials/folders/${crumb.id}`,
+        title: crumb.id < 0 ? translations.folderNotFound : crumb.name,
+        url: `materials/folders/${crumb.id < 0 ? '' : crumb.id}`,
```

Comment:
since we have the routes `materials/folders` to redirect to root folder, I think it will be better for this `url` to follow that as well i.e. if the crumb is associated with root folder, use `materials/folders` instead

Reply:
The last breadcrumb is not clickable. However, in the event that it is, I have already associated it with `materials/folders/`.

What's the issue here? Is it the trailing slash `/`? I left it there since it will make the code cleaner, and the path works with and without the trailing slash.

Reply:
Updated such that on navigating to `/materials/folders`, url will update to `/materials/folders/{rootFolderId}`.